builtin/nomad/jobspec/releaser.go

```diff
@@ -0,0 +1,547 @@
+	config ReleaserConfig
+}
+
+// Config implements Configurable
+func (r *Releaser) Config() (interface{}, error) {
+	return &r.config, nil
+}
+
+// ReleaseFunc implements component.ReleaseManager
+func (r *Releaser) ReleaseFunc() interface{} {
+	return r.Release
+}
+
+// DestroyFunc implements component.Destroyer
+func (r *Releaser) DestroyFunc() interface{} {
+	return r.Destroy
+}
+
+// StatusFunc implements component.Status
+func (r *Releaser) StatusFunc() interface{} {
+	return r.Status
+}
+
+func (r *Releaser) resourceManager(log hclog.Logger, dcr *component.DeclaredResourcesResp) *resource.Manager {
+	return resource.NewManager(
+		resource.WithLogger(log.Named("resource_manager")),
+		resource.WithValueProvider(r.getNomadClient),
+		resource.WithDeclaredResourcesResp(dcr),
+		resource.WithResource(resource.NewResource(
+			resource.WithName(rmResourceJobName),
+			resource.WithState(&Resource_Job{}),
+			resource.WithCreate(r.resourceJobCreate),
+			resource.WithDestroy(r.resourceJobDestroy),
+			resource.WithStatus(r.resourceJobStatus),
+			resource.WithPlatform("nomad"),
+			resource.WithCategoryDisplayHint(sdk.ResourceCategoryDisplayHint_INSTANCE_MANAGER),
+		)),
+	)
+}
+
+// getNomadClient provides
+// the client connection used by resources to interact with Nomad.
+func (r *Releaser) getNomadClient() (*nomadClient, error) {
+	// Get our client
+	client, err := api.NewClient(api.DefaultConfig())
+	if err != nil {
+		return nil, err
+	}
+	return &nomadClient{
+		NomadClient: client,
+	}, nil
+}
+
+func (r *Releaser) resourceJobCreate(
+	ctx context.Context,
+	log hclog.Logger,
+	target *Deployment,
+	result *Release,
+	state *Resource_Job,
+	client *nomadClient,
+	st terminal.Status,
+) error {
+	// Set up clients
+	jobClient := client.NomadClient.Jobs()
+	deploymentClient := client.NomadClient.Deployments()
+
+	st.Update("Getting job...")
+	jobs, _, err := jobClient.PrefixList(target.Name)
+	if err != nil {
+		return status.Errorf(codes.Aborted, "Unable to fetch Nomad jobs: %s", err.Error())
+	} else if target.Name != jobs[0].ID {
+		return status.Errorf(codes.Aborted, "Job not found: %s", err.Error())
+	}
+
+	q := &api.QueryOptions{
+		Namespace: jobs[0].JobSummary.Namespace,
+	}
+
+	job, _, err := jobClient.Info(jobs[0].ID, q)
+
+	if err != nil {
+		return status.Errorf(codes.Aborted, "Unable to fetch Nomad job: %s", err.Error())
+	}
+
+	st.Update("Getting latest deployments for job")
+	deploy, _, err := jobClient.LatestDeployment(*job.ID, q)
+	if err != nil {
+		return status.Errorf(codes.Aborted, "Unable to fetch latest deployment for Nomad job: %s", err.Error())
+	} else if deploy == nil {
+		st.Update("No active deployment for Nomad job")
+		return err
+	}
+
+	canaryDeployment := false
+	groupsToPromote := make([]string, len(deploy.TaskGroups))
+	for taskGroupName, taskGroup := range deploy.TaskGroups {
+		if r.config.Groups != nil {
+			if isElementExist(r.config.Groups, taskGroupName) && taskGroup.DesiredCanaries != 0 {
```

Comment:
You use `> 0` when updating the task group's canary, so for consistency, can we check `>` here too? Or is there a specific reason for checking `!= 0`?
```suggestion
			if isElementExist(r.config.Groups, taskGroupName) && taskGroup.DesiredCanaries != 0 {
```

Reply:
@krantzinator `> 0` makes more sense here, thanks!